android/libraries/rib-coroutines-test/src/main/kotlin/com/uber/rib/core/TestRibCoroutineScopes.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2022. Uber Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.uber.rib.core
+
+import com.uber.autodispose.ScopeProvider
+import com.uber.autodispose.coroutinesinterop.autoDispose
+import io.reactivex.Completable
+import io.reactivex.CompletableSource
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.test.TestCoroutineScope
+import kotlin.coroutines.CoroutineContext
+
+@ExperimentalCoroutinesApi
+/**
+ * Allows overriding [ScopeProvider.coroutineScope] with a [TestCoroutineScope] for testing
+ */
+val ScopeProvider.testCoroutineScopeOverride: TestCoroutineScope?
```

Comment:
@jbarr21 @idanakav This is the test utils class for the Scopes. There were multiple use-cases where having the scope be overridden would make since, and it would be harder to test a lifecycle based scope with your interactor otherwise. So I've added this along with some test example code.